### PR TITLE
Add py-fann2 recipe

### PR DESCRIPTION
**Pull request by smithsp**
If the import test fails, I may need help.  The import test is failing on my system during the build process, but if I forgo the test and install the package, then it seems to succeed.

### CODE REVIEW

**Comment by smithsp** [comment]
This recipe depends on #4527

File: recipes/py-fann2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package:
+  name: py-fann2
+  version: "1.1.0"
+
+source:
```
Comment (CJ-Wright) [inline recipes/py-fann2/meta.yaml]: Please use the standard syntax from the example

File: recipes/py-fann2/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  git_rev: c5b353587dfc68022a2e1f1b7a4ecfdabbb8d485
+  git_url:  git@github.com:smithsp/fann2.git
+
+build:
+  skip: True  # [win]
+  number: 0
+  # noarch_python: True
```
Comment (CJ-Wright) [inline recipes/py-fann2/meta.yaml]: please remove these comments

File: recipes/py-fann2/meta.yaml (hunk 1 above)
Comment (smithsp) [reply recipes/py-fann2/meta.yaml]: @CJ-Wright I patched the source by way of an extra git commit on my fork of their project.  Is that frowned on?
When you say to use the standard syntax, do you mean to specify the version and such as variables at the top that are reused throughout the `meta.yaml` file?

Comment (CJ-Wright) [reply recipes/py-fann2/meta.yaml]: Yes  https://conda-forge.org/docs/meta.html#build-from-tarballs-not-repos
You can put patches in the recipe. https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#patches

**Comment by smithsp** [comment]
@CJ-Wright I think that I have addressed your requests, and this branch is now passing.  Thanks for the feedback and patience.  Let me know if you see something more that needs to be changed.

File: recipes/py-fann2/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url:  https://github.com/FutureLinkCorporation/{{ name }}/archive/{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+  patches:
+    - setup.py.patch
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  # Skip the win build because fann2 doesn't build there yet
```
Comment (CJ-Wright) [inline recipes/py-fann2/meta.yaml]: Would you mind removing this comment? When you commit please use `[ci skip]` `[circle skip]` ` [travis skip]` to skip the CIs. https://github.com/conda-forge/staged-recipes/issues/1148

File: recipes/py-fann2/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - python
+    - fann2
+
+test:
+  imports:
+    - fann2
+
+about:
+  home: https://github.com/FutureLinkCorporation/fann2
+  license: LGPL2.1
```
Comment (CJ-Wright) [inline recipes/py-fann2/meta.yaml]: `LGPL-2.1` from https://opensource.org/licenses/alphabetical

**Comment by smithsp** [comment]
Thanks @CJ-Wright for your help.

**Comment by CJ-Wright** [comment]
Np, good luck with the feedstock!

**Pull request merged**